src/Umbraco.Web.Common/Builder/UmbracoBuilderExtensions.cs

```diff
@@ -54,8 +54,10 @@ public static class UmbracoBuilderExtensions
             if (config is null) throw new ArgumentNullException(nameof(config));
 
             var loggingConfig = new LoggingConfiguration(Path.Combine(webHostEnvironment.ContentRootPath, "umbraco", "logs"));
-            services.AddSingleton<ILoggingConfiguration>(loggingConfig);
-            services.AddLogger(loggingConfig, config);
+
+            var hostingSettings = config.GetSection(Core.Constants.Configuration.ConfigHosting).Get<HostingSettings>() ?? new HostingSettings();
+            var hostingEnvironment = new AspNetCoreHostingEnvironment(new OptionsMonitorAdapter<HostingSettings>(hostingSettings), webHostEnvironment);
```

Comment:
What's this for?

Reply:
TypeFinder writes a hash to temp dir, this is used when TypeLoader.GetTypesInternalLocked is called to detect whether to TypeFind again.

It's a mess, could chuck the hash in IRuntimeCache or something instead but i'm just trying to do the minimum possible to get a TypeLoader on UmbracoBuilder

Reply:
Thanks... I'll have a deeper look at this. Seems like another hack we'll need to fix (_existing work, not yours!_) in the future.

Reply:
There's a lot of that.

Reply:
Why confused?

Reply:
Ah, no... that was meant to be sadface.

Reply:
Is that not a "premature optimization", what impact would it have if scanning occurred each restart?

Reply:
> Unless this instance is the one registered as IHostingEnvironment in DI.

No that happens in AddUmbracoCore

```csharp
builder.Services.AddUnique<IHostingEnvironment, AspNetCoreHostingEnvironment>();
```

Reply:
> Is that not a "premature optimization", what impact would it have if scanning occurred each restart?

This was done years ago as a calculated optimization and followed the same patterns that netframework MVC uses to prevent rescanning types on startup. That said, a lot has changed over the last couple years and even more with netcore. I will make a note to create a benchmark to see what perf differences there is for scanning the IDiscoverable types vs loading them in from a known list (the hash is saved along with the known list of types found).

Reply:
If it is required, perhaps the mechanism could be abstracted (IDistributedCache with persistent store? something else), or have a decorator class TypeLoader wrapped by CachingTypeLoader.